src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.NamedTypeSymbolKey.cs

```diff
@@ -13,9 +13,12 @@ private static class NamedTypeSymbolKey
         {
             public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
             {
-                visitor.WriteString(symbol.MetadataName);
                 visitor.WriteSymbolKey(symbol.ContainingSymbol);
+                visitor.WriteString(symbol.Name);
                 visitor.WriteInteger(symbol.Arity);
+                visitor.WriteString(symbol.IsFile
+                    ? symbol.DeclaringSyntaxReferences[0].SyntaxTree.FilePath
+                    : null);
```

Comment:
this PR depends on a `.IsFile` property being available for us to query.

Reply:
Did the API meeting finalize that name? Do we want to call that IsFileScoped or something else? I admit it's a bit strange seeing a question which I first read "is this symbol a file?"